.github/workflows/src/verify-run-status.js

```diff
@@ -0,0 +1,174 @@
+import { extractInputs } from "./context.js";
+import { PER_PAGE_MAX } from "./github.js";
+
+/**
+ * @typedef {import('@octokit/plugin-rest-endpoint-methods').RestEndpointMethodTypes} RestEndpointMethodTypes
+ * @typedef {RestEndpointMethodTypes["checks"]["listForRef"]["response"]["data"]["check_runs"][number]} CheckRun
+ * @typedef {RestEndpointMethodTypes["actions"]["listWorkflowRunsForRepo"]["response"]["data"]["workflow_runs"][number]} WorkflowRun
+ */
+
+/**
+ * Given the name of a completed check run name and a completed workflow, verify 
+ * that both have the same conclusion. If conclusions are different, fail the 
+ * action.
+ * @param {import('github-script').AsyncFunctionArguments} AsyncFunctionArguments
+ */
+export async function verifyRunStatus({ github, context, core }) {
+  const checkRunName = process.env.CHECK_RUN_NAME;
+  if (!checkRunName) {
+    throw new Error("CHECK_RUN_NAME is not set");
+  }
+
+  const workflowName = process.env.WORKFLOW_NAME;
+  if (!workflowName) {
+    throw new Error("WORKFLOW_NAME is not set");
+  }
+
+  // Exit early when context is a check_run event and the check run does not
```

Comment:
Can this be moved into the YAML, so the job can be skipped before it even needs an agent?  Example:

https://github.com/Azure/azure-rest-api-specs/blob/b65cbab6abfdad4135f411bb1aaaeb76d27e4e70/.github/workflows/sdk-breaking-change-labels.yaml#L12

Reply:
Got it working: https://github.com/Azure/azure-rest-api-specs/pull/33766/files#diff-661cd7f3f35c083685b81e598401a6ffa7bf3bffb7af63129db68f127f45e90bR22